File: Example 12.go

package main

import (
	"fmt"
	"strconv"
)

func generateData(n int) []string {
	baseNumber := "082"
	var customers []string
	var mobileNumber string

	for i := 0; i < n; i++ {
		mobileNumber = baseNumber + strconv.Itoa(i+1)
		customers = append(customers, mobileNumber)
	}
	return customers
}

func sendPromoDiscount(input []string) {
	for _, customer := range input {
		fmt.Println("Sending Promo to " + customer)
	}
	fmt.Println("Finished sending Promo to " + strconv.Itoa(len(input)) + " Customers")

	for _, customer := range input {
		fmt.Println("Sending Discount to " + customer)
	}
	fmt.Println("Finished sending Discount to " + strconv.Itoa(len(input)) + " Customers")
}

func main() {
	data := generateData(1000)
	sendPromoDiscount(data)
}

File: Example 12.js
function generateData(n){
    const baseNumber = '082' ;
    let customers = [];
    let mobileNumber;

    for (let i = 0 ; i < n; i++){
        mobileNumber = baseNumber + (String(i).padStart(9, '0'));
        customers.push(mobileNumber)
    }
    return customers;
}

function sendPromoDiscount(input){
    for(let i = 0; i < input.length; i++){
        console.log('Sending Promo to ' + input[i]);
    }
    console.log('Its Finished to send Promo to ' + input.length + 'Customers');

    for (let i = 0; i < input.length; i++){
        console.log('Sending Discount to ' + input[i]);
    }
    console.log('Its Finished to send Discount to ' + input.length + ' Customers');
}

let data = generateData(1000);
sendPromoDiscount(data);

